### DISCUSSION TITLE
GFF3 track with multiple feature types

### DISCUSSION THREAD

**Discussion by Thys3Potgieter** [GMOD/jbrowse-components#2922] [General] [answered]
Dear community,

I am currently loading genome annotations obtained from en embl format file, converted to gff3 using emboss seqret. The gff3 contains multiple feature types, such as CDS, gene, tRNA. I would like to colour these different features differently, and ideally separate them into separate tracks.

What is the best way to achieve this? Is this done in config.json per feature type, or is there a tool that can split a gff3 into separate                                           files for each feature?

Much appreciated in advance!

Best
Thys

**Comment by Thys3Potgieter** [comment, marked as answer]
I have written a code to export these as separate tracks - but realize having one track but with different styling for feature type will be really nice. ie. CDS, gene,protein. Can anyone please let me know if this is available and where in the docs I can find out about it? Thanks!!

**Reply by cmdcolin** [reply]
the basic idea of this would be adding a color callback using jexl. I want to add a FAQ section on this at some point

here are some possibly points of reference  https://jbrowse.org/jb2/docs/config_guide/#configuration-callbacks

automating it using add-track 

https://github.com/GMOD/jbrowse-components/discussions/2700
https://github.com/GMOD/jbrowse-components/discussions/2772

here is a JEXL expression that can be used to color by type

cast({match:'green',CDS:'purple'})[get(feature,'type')]

**Reply by Thys3Potgieter** [reply]
Dear @cmdcolin just something more on this topic. I have gotten the coloring and width configurations to work like this as per your guidelines,thanks!
For color this works:
"jexl:cast({open_reading_frame:'green',polypeptide:'purple'})[get(feature,'type')]"

But for showLabels this doesn't work:
"jexl:cast({open_reading_frame:true,polypeptide:false})[get(feature,'type')]"

Is there any obvious reason why the same method doesn't work for showLabels ?
I get the error:
TypeError: Cannot read properties of undefined (reading 'get')

**Reply by cmdcolin** [reply]
sorry I haven't gotten back to you on this. I will have to see but showLabels might be a binary on off switch?

**Comment by Thys3Potgieter** [comment]
Thanks!

**Comment by Thys3Potgieter** [comment]
Thanks Colin I think so. Not a priority for me at the moment.
